### ISSUE TITLE
add "import-bundle" package

### ISSUE THREAD

**Issue by warner** [Agoric/agoric-sdk#739]
I'm going to add a package named `import-bundle`, which is the evaluation-time receiver for the data artifacts created by `bundle-source`. The `import-bundle` package will have an export named `importBundle`, that is an async function which takes a bundle (exactly as emitted by `bundleSource()`) and yields a module namespace. It will have an option for extra endowments.

For now, this will be used by SwingSet, to load the kernel bundle and all static vats. Eventually it will help factor out this code in zoe (although this might instead turn into a dynamic vat creation call):

https://github.com/Agoric/agoric-sdk/blob/d39c827866cab9ad8e9d114353a8ff5feb000843/packages/zoe/src/evalContractCode.js#L12-L23

`import-bundle` will depend upon `ses-adapter` (#724), so it should be usable from outside a SwingSet/SES environment.

**Comment by warner** [comment]
current status: I've disabled the `transforms` test in my draft PR (#762), and test pass locally. They don't pass in CI, and I don't know why, because `tap` tries+fails to set `Error.stackTraceLimit` as it attempts to render the failure. I'm using `lockdown()` with `noTameError` (because if I don't, the test doesn't run at all, and I don't know why because it prints a megabyte or two of minified code for no discernable reason), but I think `Error` is still frozen somehow.

@erights is working on this in https://github.com/Agoric/SES-shim/issues/8

**Comment by warner** [comment]
I realized a problem yesterday, and have a workaround.

`ses-adapter`, at import time, looks at the global environment for a `Compartment` (which will be present if some earlier `import` called the SES `lockdown()` function, or if we're in a future JS world where `Compartment` is standardized, like maybe XS). It exports that `Compartment` if it can, otherwise it exports a shimmed version (which may not be as secure, particularly if the primordials haven't been tamed). And at the moment, it exports a non-functional stub, because we don't yet have a separate `compartment-shim` package to work with.

Because this check happens at *import* time, any non-import-time (non-top-level) call to `lockdown()` will be too late to satisfy it. In particular, if swingset imports `import-bundle` (to load the kernel into a new Compartment), and `import-bundle` gets its `Compartment` by importing `ses-adapter`, and swingset puts off calling `lockdown()` until someone calls `buildVatController()`, then import-bundle will get the non-functional Compartment, and cannot do its job.

The short term workaround for this will be to have `import-bundle` use `Compartment` from the global environment, rather than getting it from `ses-adapter`. With this change, import-bundle *only* works in a SES environment, but that environment can be set up after import time, like during `buildVatController()`. It won't work outside of SES, such as when Zoe imports `import-bundle` to load contract code. Zoe doesn't do that yet (it uses `@agoric/evaluate` and expects a single string), but the purpose of `import-bundle` is to enable that, so we'll need it sooner or later.

The long term plan we came up with, when @michaelfig and I worked out `ses-adapter`, is for swingset to require a pre-loaded SES environment. All applications and test programs will need to create a short module named e.g. `install-ses.js`, which imports `lockdown()` and calls it at the top-level. They must import this (for side-effects, e.g. `import './install-ses.js'`) *before* they import swingset, or zoe, or anything that might reach `import-bundle- or `ses-adapter`. Michael recommended that `install-ses.js` get imported before everything except some vetted shims that really need to modify the pre-hardened primordials.

In that world, swingset won't call `lockdown()`: it can just assume `Compartment` and `harden` are on the global (or maybe available as built-in modules, which I think is where the SES API is headed). And `import-bundle` can go back to using `ses-adapter`, which means Zoe can switch to using `import-bundle` and still have unit tests work outside of SES.
